src/shell-integration/bash/ghostty.bash

```diff
@@ -117,25 +137,7 @@ function __ghostty_precmd() {
       if [[ "$GHOSTTY_SHELL_INTEGRATION_NO_SUDO" != "1" ]] && [[ -n "$TERMINFO" ]]; then
         # Wrap `sudo` command to ensure Ghostty terminfo is preserved
         # shellcheck disable=SC2317
-        sudo() {
-          builtin local sudo_has_sudoedit_flags="no"
-          for arg in "$@"; do
-            # Check if argument is '-e' or '--edit' (sudoedit flags)
-            if [[ "$arg" == "-e" || $arg == "--edit" ]]; then
-              sudo_has_sudoedit_flags="yes"
-              builtin break
-            fi
-            # Check if argument is neither an option nor a key-value pair
-            if [[ "$arg" != -* && "$arg" != *=* ]]; then
-              builtin break
-            fi
-          done
-          if [[ "$sudo_has_sudoedit_flags" == "yes" ]]; then
-            builtin command sudo "$@";
-          else
-            builtin command sudo TERMINFO="$TERMINFO" "$@";
-          fi
-        }
+        alias sudo="sudo_function"
```

Comment:
I'll also have to think a little bit about the differences between using an `alias` versus function override.